e2e/interchaintestv8/e2esuite/parallel.go

```diff
@@ -0,0 +1,161 @@
+package e2esuite
+
+import (
+	"fmt"
+	"sync"
+)
+
+// ParallelTask represents a named task that can be executed concurrently
+type ParallelTask struct {
+	Name string
+	Run  func() error
+}
```

Comment:
I wonder if each task should define a cleanup function, this way, when one fails, we can clean up? Maybe they should define a `Stop()` method as well. I don't want it to be very complex though, so maybe this is ok